transcode/transcode.go

```diff
@@ -300,6 +320,52 @@ func RunTranscodeProcess(transcodeRequest TranscodeSegmentRequest, streamName st
 	return outputs, segmentsCount, nil
 }
 
+func uploadMp4Files(requestID string, basePath *url.URL, mp4OutputFiles []string, rendition string) ([]video.OutputVideoFile, error) {
+	var mp4OutputsPre []video.OutputVideoFile
+	// e. Upload all mp4 related output files
+	for _, o := range mp4OutputFiles {
+		var filename string
+		mp4OutputFile, err := os.Open(o)
+		defer os.Remove(mp4OutputFile.Name())
+		if err != nil {
+			return []video.OutputVideoFile{}, fmt.Errorf("failed to open %s to upload: %s", o, err)
+		}
+		ext := filepath.Ext(mp4OutputFile.Name())
+		if ext == ".mp4" {
+			filename = fmt.Sprintf("%s.mp4", rendition)
+		} else {
+			filename = filepath.Base(mp4OutputFile.Name())
+		}
+		err = backoff.Retry(func() error {
+			return clients.UploadToOSURL(basePath.String(), filename, bufio.NewReader(mp4OutputFile), UPLOAD_TIMEOUT)
+		}, clients.UploadRetryBackoff())
+		if err != nil {
+			return []video.OutputVideoFile{}, fmt.Errorf("failed to upload %s: %s", mp4OutputFile.Name(), err)
+		}
+		mp4Out := video.OutputVideoFile{
+			Type:     "mp4",
+			Location: basePath.JoinPath(filename).String(),
+		}
+		mp4OutputsPre = append(mp4OutputsPre, mp4Out)
+	}
+	return mp4OutputsPre, nil
+}
+
+// getMp4OutputType checks the target url of the MP4 or Fragmented-MP4
+// output location and returns an *url.URL along with a boolean to
+// indicate that specific output type (mp4 or f-mp4) has been enabled.
+func getMp4OutputType(targetUrl string) (*url.URL, bool, error) {
+	if len(targetUrl) != 0 {
```

Comment:
nit: Could you invert the if statement and return early to avoid the extra indentation?